include/framework/Checks.hpp

```diff
@@ -19,50 +19,45 @@
-  }
-}
-
-#ifndef __PRETTY_FUNCTION__
-#define __PRETTY_FUNCTION__ __FUNCTION__
-#endif
-
-// macro to easily check for GL errors
-#define CHECK_GL_ERROR() getGLError(__LINE__, __FILE__, __PRETTY_FUNCTION__);
+inline std::map<int, std::string> NVMEDIA_ERROR_TO_STRING = { { 0, "NVMEDIA_STATUS_OK" },
+                                                              { 1, "NVMEDIA_STATUS_BAD_PARAMETER" },
+                                                              { 2, "NVMEDIA_STATUS_PENDING" },
+                                                              { 3, "NVMEDIA_STATUS_TIMED_OUT" },
+                                                              { 4, "NVMEDIA_STATUS_OUT_OF_MEMORY" },
+                                                              { 5, "NVMEDIA_STATUS_NOT_INITIALIZED" },
+                                                              { 6, "NVMEDIA_STATUS_NOT_SUPPORTED" },
+                                                              { 7, "NVMEDIA_STATUS_ERROR" },
+                                                              { 8, "NVMEDIA_STATUS_NONE_PENDING" },
+                                                              { 9, "NVMEDIA_STATUS_INSUFFICIENT_BUFFERING" },
+                                                              { 10, "NVMEDIA_STATUS_INVALID_SIZE" },
+                                                              { 11, "NVMEDIA_STATUS_INCOMPATIBLE_VERSION" },
+                                                              { 13, "NVMEDIA_STATUS_UNDEFINED_STATE" },
+                                                              { 14, "NVMEDIA_STATUS_PFSD_ERROR" } };
 
 inline void getDateString(char* buf, size_t length)
 {
-  time_t now = ::time(0);
-  struct tm* calendar = localtime(&now);
+  time_t now = ::time(nullptr);
+  struct tm const* calendar = localtime(&now);
   strftime(buf, length, "[%Y-%m-%d %X] ", calendar);
 }
 
-#define CHECK_DW_ERROR_ROS(x)                                                                                          \
-  {                                                                                                                    \
+#define CHK_DW(x)                                                                                                      \
+  do {                                                                                                                 \
     dwStatus result = x;                                                                                               \
     if (result != DW_SUCCESS) {                                                                                        \
       char buf[80];                                                                                                    \
       getDateString(buf, 80);                                                                                          \
+      std::cout << std::string(buf) + std::string("DW Error ") + dwGetStatusName(result) +                             \
```

Comment:
Is this cout planned or just a leftover? Other similar macros here do not print anything before throwing exception.

Reply:
It was a leftover, removed it.